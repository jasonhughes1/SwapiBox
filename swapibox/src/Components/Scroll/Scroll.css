.scroll-container {
  display: flex;
  flex-direction: column;
  justify-content: center; }

.opening-crawl,
.opening-title,
.opening-releaseDate {
  width: 50%;
  position: relative;
  top: -100px;
  transform-origin: 90% 90%;
  animation: crawl 30s infinite linear; }

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0); }
  100% {
    top: -1500px;
    transform: rotateX(25deg) translateZ(-2500deg); } }
