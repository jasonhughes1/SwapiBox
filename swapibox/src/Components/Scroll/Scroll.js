import React from 'react';
import css from './Scroll.css';

const Scroll = ({data, opening}) => {
const text = data[opening].Opening;
const filmTitle = data[opening].Title;
const releaseDate = data[opening].Release;

  return (
    <div>
      <div className='crawl-container'>
        <div className='fade'>
        </div>
        <div className='star-wars'>
          <div className='crawl'>
            <p className='crawl-text'>{text}</p>
            <p className='film-title'>{filmTitle}</p>
            <p className='release-date'>{releaseDate}</p>
          </div>
        </div>
      </div>
    </div>
  );
};


export default Scroll;
