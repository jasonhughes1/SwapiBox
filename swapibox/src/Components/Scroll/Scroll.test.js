import React from 'react';
import Scroll from './Scroll.js';
import { shallow } from 'enzyme';


let renderedScroll;

let mockData = [
  {
    Opening: "It is a period of civil was",
    Title: "A New Hope",
    Release: "1977-05-25"
  }, {
    Opening: "It is a period of civil was",
    Title: "A New Hope",
    Release: "1977-05-25"
  }, {
    Opening: "It is a period of civil was",
    Title: "A New Hope",
    Release: "1977-05-25"
  }, {
    Opening: "It is a period of civil was",
    Title: "A New Hope",
    Release: "1977-05-25"
  }, {
    Opening: "It is a period of civil was",
    Title: "A New Hope",
    Release: "1977-05-25"
  }, {
  }, {
    Opening: "It is a period of civil was",
    Title: "A New Hope",
    Release: "1977-05-25"
  }, {
    Opening: "It is a period of civil was",
    Title: "A New Hope",
    Release: "1977-05-25"
  }];

let mockOpening = 1;

describe('Scroll Tests', () => {
  beforeEach(() => {
    renderedScroll = shallow(
      <Scroll
        data={mockData}
        opening={mockOpening}
      />);
  });

  it('should render correctly', () => {
    expect(renderedScroll).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(renderedScroll).toMatchSnapshot();
  });
});
