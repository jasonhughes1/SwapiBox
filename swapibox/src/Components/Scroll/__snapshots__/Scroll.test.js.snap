// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll Tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroll
    data={
        Array [
            Object {
              "Opening": "It is a period of civil was",
              "Release": "1977-05-25",
              "Title": "A New Hope",
            },
            Object {
              "Opening": "It is a period of civil was",
              "Release": "1977-05-25",
              "Title": "A New Hope",
            },
            Object {
              "Opening": "It is a period of civil was",
              "Release": "1977-05-25",
              "Title": "A New Hope",
            },
            Object {
              "Opening": "It is a period of civil was",
              "Release": "1977-05-25",
              "Title": "A New Hope",
            },
            Object {
              "Opening": "It is a period of civil was",
              "Release": "1977-05-25",
              "Title": "A New Hope",
            },
            Object {},
            Object {
              "Opening": "It is a period of civil was",
              "Release": "1977-05-25",
              "Title": "A New Hope",
            },
            Object {
              "Opening": "It is a period of civil was",
              "Release": "1977-05-25",
              "Title": "A New Hope",
            },
          ]
    }
    opening={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="crawl-container"
>
        <div
                className="fade"
        />
        <div
                className="star-wars"
        >
                <div
                        className="crawl"
                >
                        <p
                                className="crawl-text"
                        >
                                It is a period of civil was
                        </p>
                        <p
                                className="film-title"
                        >
                                A New Hope
                        </p>
                        <p
                                className="release-date"
                        >
                                1977-05-25
                        </p>
                </div>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <div
            className="star-wars"
>
            <div
                        className="crawl"
            >
                        <p
                                    className="crawl-text"
                        >
                                    It is a period of civil was
                        </p>
                        <p
                                    className="film-title"
                        >
                                    A New Hope
                        </p>
                        <p
                                    className="release-date"
                        >
                                    1977-05-25
                        </p>
            </div>
</div>,
        ],
        "className": "crawl-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <p
                            className="crawl-text"
              >
                            It is a period of civil was
              </p>
              <p
                            className="film-title"
              >
                            A New Hope
              </p>
              <p
                            className="release-date"
              >
                            1977-05-25
              </p>
</div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="crawl-text"
>
                  It is a period of civil was
</p>,
                <p
                  className="film-title"
>
                  A New Hope
</p>,
                <p
                  className="release-date"
>
                  1977-05-25
</p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "It is a period of civil was",
                  "className": "crawl-text",
                },
                "ref": null,
                "rendered": "It is a period of civil was",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A New Hope",
                  "className": "film-title",
                },
                "ref": null,
                "rendered": "A New Hope",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1977-05-25",
                  "className": "release-date",
                },
                "ref": null,
                "rendered": "1977-05-25",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl-container"
>
          <div
                    className="fade"
          />
          <div
                    className="star-wars"
          >
                    <div
                              className="crawl"
                    >
                              <p
                                        className="crawl-text"
                              >
                                        It is a period of civil was
                              </p>
                              <p
                                        className="film-title"
                              >
                                        A New Hope
                              </p>
                              <p
                                        className="release-date"
                              >
                                        1977-05-25
                              </p>
                    </div>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fade"
/>,
            <div
              className="star-wars"
>
              <div
                            className="crawl"
              >
                            <p
                                          className="crawl-text"
                            >
                                          It is a period of civil was
                            </p>
                            <p
                                          className="film-title"
                            >
                                          A New Hope
                            </p>
                            <p
                                          className="release-date"
                            >
                                          1977-05-25
                            </p>
              </div>
</div>,
          ],
          "className": "crawl-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fade",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="crawl"
>
                <p
                                className="crawl-text"
                >
                                It is a period of civil was
                </p>
                <p
                                className="film-title"
                >
                                A New Hope
                </p>
                <p
                                className="release-date"
                >
                                1977-05-25
                </p>
</div>,
              "className": "star-wars",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="crawl-text"
>
                    It is a period of civil was
</p>,
                  <p
                    className="film-title"
>
                    A New Hope
</p>,
                  <p
                    className="release-date"
>
                    1977-05-25
</p>,
                ],
                "className": "crawl",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "It is a period of civil was",
                    "className": "crawl-text",
                  },
                  "ref": null,
                  "rendered": "It is a period of civil was",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                    "className": "film-title",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1977-05-25",
                    "className": "release-date",
                  },
                  "ref": null,
                  "rendered": "1977-05-25",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
